private-templates-service/adaptivecards-templating-service/src/TemplateServiceClient.ts

```diff
@@ -0,0 +1,170 @@
+import { ClientOptions } from "./IClientOptions";
+import express, { Request, Response, NextFunction, Router } from "express";
+import { check, validationResult } from "express-validator";
+import { AuthenticationProvider } from "."; 
+import { TemplateError, ApiError } from "./api/TemplateError";
+import { StorageProvider } from ".";
+import { ITemplate, JSONResponse, ITemplateInstance } from ".";
+
+export class TemplateServiceClient {
+
+    private static storageProvider: StorageProvider;
+    private static authProvider: AuthenticationProvider;
+
+    /**
+     * @public
+     * Initialize database if not already running
+     * @param {ClientOptions} clientOptions - storage provider and auth provider options
+     */
+    public static init(clientOptions: ClientOptions): TemplateServiceClient {
+        // TODO: add db setup step once mongo adapter is added
+
+        if (clientOptions.storageProvider === undefined) {
+            const error = new Error();
+            error.name = "Missing Storage Provider";
+            error.message = "Please provide a storage provider.";
+            throw error;
+        }
+
+        if (clientOptions.authenticationProvider === undefined) {
+            const error = new Error();
+            error.name = "Missing Authentication Provider";
+            error.message = "Please provide an authentication provider";
+            throw error;
+        }
+
+        TemplateServiceClient.storageProvider = clientOptions.storageProvider;
+        TemplateServiceClient.authProvider = clientOptions.authenticationProvider;
```

Comment:
does typescript allow you to access to the private fields through class name? I would expect this keyword here. that is weird to me :-/

Reply:
Changed to use this!